semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -49,72 +58,53 @@ def parse(span_kind_value):
         return kind_map.get(span_kind_value)
 
 
-class SemanticConventionType(Enum):
-    SPAN = 1
-    RESOURCE = 2
-    METRIC = 3
-
-    @staticmethod
-    def parse(type_value):
-        # Gracefully transition to the new types
-        if type_value is None:
-            return SemanticConventionType.SPAN
-        enum_map = {
-            "span": SemanticConventionType.SPAN,
-            "resource": SemanticConventionType.RESOURCE,
-        }
-        return enum_map.get(type_value)
+def parse_semantic_convention_type(type_value):
+    # Gracefully transition to the new types
+    if type_value is None:
+        return SpanSemanticConvention
+    enum_map = {
+        "span": SpanSemanticConvention,
+        "resource": ResourceSemanticConvention,
+        "metric": MetricSemanticConvention,
+        "units": UnitSemanticConvention,
+    }
+    return enum_map.get(type_value)
 
 
-@dataclass
-class SemanticConvention:
+class SemanticConvention(ValidatableYamlNode):
     """ Contains the model extracted from a yaml file
     """
 
-    semconv_id: str
-    type: SemanticConventionType
-    brief: str
-    note: str
-    prefix: str
-    extends: str
-    span_kind: SpanKind
-    attrs_by_name: "Dict[str, SemanticAttribute]"
-    constraints: "Set[Union[Include, AnyOf]]"
-    _position: List[int]
+    def __init__(self, group):
+        super(SemanticConvention, self).__init__(group)
+
+        self.semconv_id = self.id
+        self.note = group.get("note", "").strip()
+        self.prefix = group.get("prefix", "")
```

Comment:
Although it is improbable that users will use the `:>` notation, I would still keep the `stip` to avoid newlines in the prefix.
```suggestion
        self.prefix = group.get("prefix", "").strip()
```